### PR TITLE
Iterate upon error message styling

### PR DESCRIPTION
**Pull request by sigvef**
Before:

image

---

After:

image

### CODE REVIEW

File: nin/frontend/app/scripts/filters/stackFormat.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      var line = lines[i];
+      line = line.trim();
+      line = line.replace(/</g, '&lt;');
+      line = line.replace(/>/g, '&gt;');
+      line = line.replace(/^at/, '<span class="muted">↳</span>');
+      line = line.replace(/https?:\/\/[^\/]+:8000/, '<em>nin</em>');
+      line = line.replace(/https?:\/\/[^\/]+:9000/, '<em>demo</em>');
+      line = line.replace(/(\(.*)/, '<span class="muted">$1</span>');
+      html += '\n' + line;
+    }
```
Comment (stianjensen) [inline nin/frontend/app/scripts/filters/stackFormat.js]: Is this format equal in all browsers or specific to Chrome?

Comment (sigvef) [inline nin/frontend/app/scripts/filters/stackFormat.js]: No idea. Do we use anything other than Chrome?

Comment (aleksanb) [inline nin/frontend/app/scripts/filters/stackFormat.js]: I usually use Firefox.

File: nin/frontend/app/styles/common.less [hunk 2]
```diff
@@ -530,8 +530,29 @@ input[type=range]:focus::-ms-fill-upper {
   padding-bottom: 10px;
   font-family: monospace;
   z-index: 999999999999999;
-  background-color: rgba(113, 75, 75, 0.6);
-  overflow-y: scroll;
```
Comment (aleksanb) [inline nin/frontend/app/styles/common.less]: Is there a way to get the error to still show the line number? In your example it has wrapped outside the screen. Would it be better if the muted part of he error could wrap over multiple lines, slightly indented?

**Comment by aleksanb** [comment]
Is there a way to get the error to still show the line number? In your example it has wrapped outside the screen. Would it be better if the muted part of he error could wrap over multiple lines, slightly indented?

File: nin/frontend/app/styles/common.less (hunk 2 above)
Comment (aleksanb) [inline nin/frontend/app/styles/common.less]: Dammit I can't seem to move or delete line notes. I've commented globally as well.

File: nin/frontend/app/scripts/filters/stackFormat.js (hunk 1 above)
Comment (cristeahub) [inline nin/frontend/app/scripts/filters/stackFormat.js]: I use Vivaldi, while that uses Chromium there are some differences, not sure if this specific format will be different though.

**Comment by aleksanb** [comment]
Making the project seems to fail now. Perhaps some linting rule's gone amok?

Running "ngmin:dist" (ngmin) task
ngminifying .tmp/concat/scripts/scripts.js, .tmp/concat/scripts/vendor.js
Warning: Line 1233: Unexpected token ILLEGAL Use --force to continue.

Aborted due to warnings.

**Comment by sigvef** [comment]
image

**Comment by aleksanb** [comment]
Currently blocked on Make failing.

**Comment by sigvef** [comment]
Making works fine on my machine. Let's compare environments:

```
$ npm list | grep ngmin
├─┬ grunt-ngmin@0.0.3
│ └─┬ ngmin@0.4.1
```

What is your output for the same command (run from the frontend folder)?

**Comment by aleksanb** [comment]
I get the following horror:
```
├─┬ grunt-ngmin@0.0.3
│ └─┬ ngmin@0.4.1
npm ERR! extraneous: noptify /home/aleksanb/Projects/nin/nin/frontend/node_modules/noptify
npm ERR! error in /home/aleksanb/Projects/nin/nin/frontend/node_modules/noptify: ENOENT: no such file or directory, open '/home/aleksanb/Projects/nin/nin/frontend/node_modules/noptify/package.json'
npm ERR! extraneous: pngquant-bin /home/aleksanb/Projects/nin/nin/frontend/node_modules/pngquant-bin
npm ERR! error in /home/aleksanb/Projects/nin/nin/frontend/node_modules/pngquant-bin: ENOENT: no such file or directory, open '/home/aleksanb/Projects/nin/nin/frontend/node_modules/pngquant-bin/package.json'
npm ERR! extraneous: read-package-json /home/aleksanb/Projects/nin/nin/frontend/node_modules/read-package-json
npm ERR! error in /home/aleksanb/Projects/nin/nin/frontend/node_modules/read-package-json: ENOENT: no such file or directory, open '/home/aleksanb/Projects/nin/nin/frontend/node_modules/read-package-json/package.json'
npm ERR! extraneous: webpack@1.13.1 /home/aleksanb/Projects/nin/nin/frontend/node_modules/webpack
npm ERR! extraneous: fstream /home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/fstream
npm ERR! error in /home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/fstream: ENOENT: no such file or directory, open '/home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/fstream/package.json'
npm ERR! extraneous: update-notifier /home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/update-notifier
npm ERR! error in /home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/update-notifier: ENOENT: no such file or directory, open '/home/aleksanb/Projects/nin/nin/frontend/node_modules/grunt-google-cdn/node_modules/update-notifier/package.json'
```

Note that i get the same error on both master and this branch, but making works fine on master.

**Comment by aleksanb** [comment]
Errors might just be from stderr and not related to ngmin

**Comment by aleksanb** [comment]
Ah, i know the error!

**Comment by aleksanb** [comment]
What Node version do you have? I'm running 4.2.6.

**Comment by sigvef** [comment]
ping, the branch was out of date

**Comment by stianjensen** [comment]
`npm prune` should remove warnings about extraneous packages

File: nin/frontend/app/views/top-panel.html [hunk 3]
```diff
@@ -94,13 +94,17 @@
...
+    <!-- cramped formatting because of whitespace: pre -->
     <div
       class=global-error-overlay
       ng-repeat="(_, error) in globalJSErrors"
-      >
-      <pre>⛄ {{ error.context }} ⛄</pre>
-      <pre>{{ error.stack }}</pre>
-    </div>
+      ><div class="context">⛄ {{ error.context }} ⛄</div>
+<div class="bg">
```
Comment (aleksanb) [inline nin/frontend/app/views/top-panel.html]: Consider indenting.

File: nin/frontend/app/scripts/controllers/main.js [hunk 4]
```diff
@@ -240,6 +242,7 @@
 
             case 'shader':
               var indirectEval = eval;
+              $scope.fileCache[event.path] = event.content;
```
Comment (aleksanb) [inline nin/frontend/app/scripts/controllers/main.js]: You will need to add more fileCaches to intercept the original loads as well i believe.
Layers can be loaded by the 'layers' or 'layer' event afaik, which atm contain duplicated code. the 'layers' should probably just call some function shared with the 'layer' event.

Comment (aleksanb) [inline nin/frontend/app/scripts/controllers/main.js]: Disregard this, i'm smoking something.
We only send the manifest on 'layers'.

File: nin/frontend/app/views/top-panel.html (hunk 3 above)
Comment (sigvef) [inline nin/frontend/app/views/top-panel.html]: image

Comment (sigvef) [inline nin/frontend/app/views/top-panel.html]: image

Comment (aleksanb) [inline nin/frontend/app/views/top-panel.html]: Ah! I thought as much, but the comment was so far up i didn't see it.

**Pull request merged**